pilot/pkg/model/sidecar.go

```diff
@@ -216,12 +216,14 @@ func DefaultSidecarScopeForNamespace(ps *PushContext, configNamespace string) *S
 		})
 	}
 
-	for _, dr := range out.destinationRules {
-		out.AddConfigDependencies(ConfigKey{
-			Kind:      gvk.DestinationRule,
-			Name:      dr.Name,
-			Namespace: dr.Namespace,
-		})
+	for _, drList := range out.destinationRules {
+		for _, dr := range drList {
+			out.AddConfigDependencies(ConfigKey{
```

Comment:
Not related to this pr, i think from merge beginning the dependency is not right. We can ignore some drs, if they are merged in to a dr